instrumentation/logback/logback-appender-1.0/library/src/main/java/io/opentelemetry/instrumentation/logback/appender/v1_0/LoggingEventToReplay.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.logback.appender.v1_0;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.classic.spi.IThrowableProxy;
+import ch.qos.logback.classic.spi.LoggerContextVO;
+import io.opentelemetry.instrumentation.logback.appender.v1_0.internal.LoggingEventWithThreadId;
+import java.util.List;
+import java.util.Map;
+import org.slf4j.Marker;
+import org.slf4j.event.KeyValuePair;
+
+class LoggingEventToReplay implements LoggingEventWithThreadId {
+
+  private final ILoggingEvent loggingEvent;
+  private final long timeStamp;
+  private StackTraceElement[] callerData;
+  private String threadName;
+  private long threadId;
+
+  LoggingEventToReplay(
+      ILoggingEvent loggingEvent,
+      boolean captureExperimentalAttributes,
+      boolean captureCodeAttributes) {
+    this.loggingEvent = loggingEvent;
+    this.timeStamp = loggingEvent.getTimeStamp();
+    if (captureExperimentalAttributes) {
+      this.threadName = loggingEvent.getThreadName();
+      this.threadId = Thread.currentThread().getId();
```

Comment:
I only just realized that our implementation using `Thread.currentThread()` may be incorrect, or at least won't work for async appenders, which is probably why LoggingEvent already has `getThreadName()`.

maybe we should just not capture `thread.id` in logback appender and note the limitation